evm_types/block_test.go

```diff
@@ -0,0 +1,81 @@
+package evm_types
+
+import (
+	"github.com/Mamoru-Foundation/mamoru-sniffer-go/mamoru_sniffer"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func BenchmarkBlockDataCtxBuilder(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		builder := mamoru_sniffer.NewBlockchainDataCtxBuilder()
+
+		builder.AddData(NewBlockData([]Block{
+			{
+				BlockIndex:                 0,
+				Hash:                       "",
+				ParentHash:                 "",
+				StateRoot:                  "",
+				Nonce:                      0,
+				Status:                     "",
+				Timestamp:                  0,
+				BlockReward:                nil,
+				FeeRecipient:               "",
+				TotalDifficulty:            0,
+				Size:                       0,
+				GasUsed:                    0,
+				GasLimit:                   0,
+				BurntFees:                  nil,
+				PosProposedOnTime:          0,
+				PosSlot:                    0,
+				PosEpoch:                   0,
+				PosProposerIndex:           0,
+				PosSlotRootHash:            nil,
+				PosBeaconChainDepositCount: 0,
+				PosSlotGraffiti:            nil,
+				PosBlockRandomness:         nil,
+				PosRandomReveal:            nil,
```

Comment:
It's interesting, that it doesn't fail if `[]byte` is `nil`

Reply:
The zero value of a Slice is nil 
```
var nilSlice []byte
fmt.Println(nilSlice) // Output: []
fmt.Println(nilSlice == nil) // Output: true
```